rust/enums/src/main.rs

```diff
@@ -0,0 +1,205 @@
+use std::fs;
+use std::path::Path;
+
+use serde::Deserialize;
+
+#[derive(Deserialize, Debug, PartialEq)]
+enum AgeBracket {
+    Child,
+    Youth,
+    Adult,
+    Senior,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+#[allow(dead_code)]
+struct Person {
+    id: Option<u32>,
+    age: Option<u32>,
+    is_married: Option<bool>,
+    city: Option<String>,
+    state: Option<String>,
+    country: Option<String>,
+    age_bracket: Option<AgeBracket>,
+}
+
+impl Person {
+    fn set_age_bracket(&mut self) {
+        self.age_bracket = match self.age {
+            Some(a) => {
+                if a < 13 {
+                    Some(AgeBracket::Child)
+                } else if (13..=17).contains(&a) {
+                    Some(AgeBracket::Youth)
+                } else if (18..=59).contains(&a) {
+                    Some(AgeBracket::Adult)
+                } else {
+                    Some(AgeBracket::Senior)
+                }
+            }
+            None => None,
+        };
+    }
+}
+
+fn load_csv(file_path: &Path) -> Result<Vec<Person>, Box<dyn std::error::Error>> {
+    let contents = fs::read_to_string(file_path).expect("Unable to read from CSV");
+    let mut reader = csv::Reader::from_reader(contents.as_bytes());
+    let mut data: Vec<Person> = reader.deserialize().collect::<Result<_, _>>()?;
+
+    for d in &mut data {
+        d.set_age_bracket();
+    }
+
+    Ok(data)
+}
+
+fn main() {
+    let file_path = Path::new("data/persons.csv");
+    let persons = load_csv(file_path).expect("Unable to read/open CSV");
+    for person in persons {
+        println!("{:?}", person);
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
```

Comment:
Rust's default test client doesn't seem to support parameterize like `pytest` does, so I came up with [this approach](https://github.com/thedataquarry/rustinpieces/blob/944aab06702761a1477465027cf7d6d7b0455acc/rust/mock_data/src/test_main.rs#L5) in the other directory. We could just read the input CSV (same way as we do in the `pytest` suite) and make the result an iterable, and set the `age, expected` values via tuples in Rust. It also looks a lot cleaner, I think?